### PR TITLE
[REVIEW] Replaced nvtx package with conda-forge source

### LINKED ISSUE #6395: [FEA] Use `nvtx` conda package
Currently, cuDF code is annotated using internal NVTX bindings (`cudf._lib.nvtx`). For example: https://github.com/rapidsai/cudf/blob/branch-0.16/python/cudf/cudf/core/series.py#L17

There is now an `nvtx` package on [conda-forge](https://anaconda.org/conda-forge/nvtx). We should switch to just using that.

Note: docs for nvtx are not yet published, but can be found at https://github.com/NVIDIA/NVTX/tree/dev/python/

### PR DESCRIPTION
**Pull request by isVoid**
Closes #6395 

- Removed `cudf._lib.nvtx` package and replaced all usage to `nvtx`
- Added dependency to yml files.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: CHANGELOG.md
Comment (galipremsagar) [inline CHANGELOG.md:115]: Can this be added under `0.17` - I think you can start off one in your PR

File: CHANGELOG.md
Comment (isVoid) [inline CHANGELOG.md]: Should there be several lines of spaces to separate between version sections?

File: CHANGELOG.md
Comment (galipremsagar) [inline CHANGELOG.md:10]:
```suggestion


```

**Review by galipremsagar — APPROVED** [review]
Since we have removed `nvtx` from `cudf._lib`, we'll also have to get rid this(`cudf._lib.nvtx`) usage in other projects similarly. 

UCX:
https://github.com/rapidsai/ucx-py/blob/5652cb37063bd2f69020ffbd7f84e02a97fac267/ucp/utils.py#L231


Dask-cuda:
https://github.com/rapidsai/dask-cuda/blob/a23ea45bafee3176952d72e5a554f3b7e5232b00/dask_cuda/utils.py#L12

cc: @pentschev @kkraus14

**Comment by isVoid** [comment]
Also in several locations in NVTabular.

Such as https://github.com/NVIDIA/NVTabular/blob/90123625beed03d31e8cdd3b2141caa3a8c6346e/nvtabular/ops/column_similarity.py#L22

**Comment by pentschev** [comment]
Thanks @galipremsagar for the ping, I just opened PRs to update UCX-Py and Dask-CUDA, in https://github.com/rapidsai/ucx-py/pull/639 and https://github.com/rapidsai/dask-cuda/pull/406, respectively.

**Comment by jakirkham** [comment]
Adding to RAPIDS integration packages with PR ( https://github.com/rapidsai/integration/pull/141 ).

**Review by ajschmidt8 — APPROVED** [review]

**Comment by ajschmidt8** [comment]
rerun tests

**Comment by kkraus14** [comment]
Given there's multiple places depending on this currently, should we have a release with it still accessible at `cudf._lib.nvtx` by having the import in the `_lib.__init__`? Then we can remove it in 0.17 safely.

**Comment by galipremsagar** [comment]
> Given there's multiple places depending on this currently, should we have a release with it still accessible at `cudf._lib.nvtx` by having the import in the `_lib.__init__`? Then we can remove it in 0.17 safely.

+1 to deprecate `cudf._lib.nvtx`(with a deprecation warning) for `0.16` and removing it in `0.17`

**Comment by kkraus14** [comment]
> > Given there's multiple places depending on this currently, should we have a release with it still accessible at `cudf._lib.nvtx` by having the import in the `_lib.__init__`? Then we can remove it in 0.17 safely.
> 
> +1 to deprecate `cudf._lib.nvtx`(with a deprecation warning) for `0.16` and removing it in `0.17`

I'm not exactly sure how we'd add a deprecation warning that wouldn't get triggered everywhere that `_lib` is used (which is basically everywhere). Given it's in an internal namespace I think we're safe to change it at any time for any reason regardless, but if there's a nice way to throw a deprecation warning when imported that would be nice.

**Comment by ajschmidt8** [comment]
rerun tests

**Comment by jakirkham** [comment]
Yeah agree with Keith. As this is a private module, should be safe to remove at any time.

FWIW we have already fixed UCX-Py. We have a similar fix of Dask-CUDA, but are stuck with an unrelated test failure atm ( https://github.com/rapidsai/dask-cuda/pull/406#issuecomment-703683511 ).

**Comment by kkraus14** [comment]
Given we've bumped this to 0.17 (which I think is a good call), lets just remove it. If we cause breakages, so be it and they have an entire release to catch up 😄.

**Review by kkraus14 — APPROVED** [review]

File: CHANGELOG.md
Comment (harrism) [inline CHANGELOG.md:7]:
```suggestion
- PR #6413 Replace Python NVTX package with official source
```

**Review by harrism — CHANGES_REQUESTED** [review]
Just clarifying changelog.

**Review by harrism — APPROVED** [review]

**Comment by kkraus14** [comment]
rerun tests

**Comment by galipremsagar** [comment]
@isVoid I think you missed removing this `nvtx` import:

https://github.com/isVoid/cudf/blob/7407905452c02346f1b0d68cc1b50f3d237f610b/python/cudf/cudf/_lib/__init__.py#L19

**Comment by isVoid** [comment]
Done. Thanks for the catch!

**Comment by galipremsagar** [comment]
This PR(https://github.com/rapidsai/integration/pull/144) should unblock from CI failure

**Comment by jakirkham** [comment]
Just an update, we've already fixed UCX-Py and Dask-CUDA to use the Python NVTX package directly.

**Comment by kkraus14** [comment]
rerun tests

File: conda/recipes/cudf/meta.yaml
Comment (kkraus14) [inline conda/recipes/cudf/meta.yaml:44]: Do we want to constrain this to `nvtx >=0.2.1,<0.3.0a0` or are we comfortable we won't be introducing breaking changes? cc @shwina

**Review by kkraus14 — CHANGES_REQUESTED** [review]
Blocking merging until question is answered

File: conda/recipes/cudf/meta.yaml
Comment (shwina) [reply conda/recipes/cudf/meta.yaml:44]: I don't think we need to constrain it. There's no plan for a `0.3.0` any time soon

**Review by kkraus14 — APPROVED** [review]

**Pull request merged**
